### PR TITLE
Add read_files_as_table and make bulk_upload_files atomic

### PR DESCRIPTION
**Pull request by azorej**

### CODE REVIEW

File: dbxio/delta/table_commands.py [hunk 1]
```diff
@@ -37,24 +37,39 @@ def exists_table(table: Union[str, Table], client: 'DbxIOClient') -> bool:
...
+        schema_sql = f'({dbxio_table.schema.as_sql()})'
+    else:
+        schema_sql = ''
+    if replace:
+        query = 'CREATE OR REPLACE TABLE'
+    else:
+        query = 'CREATE TABLE'
+        if if_not_exists:
+            query += ' IF NOT EXISTS'
+    query += f' {dbxio_table.safe_table_identifier} {schema_sql}'
```
Comment (NikitaYurasov) [inline dbxio/delta/table_commands.py:53]:
```suggestion
    schema_sql = dbxio_table.schema.as_sql() if include_schema and dbxio_table.schema else ''
    query = 'CREATE OR REPLACE TABLE' if replace else f'CREATE TABLE {"IF NOT EXISTS" if if_not_exists else ""}'
    query += f' {dbxio_table.safe_table_identifier} ({schema_sql})'
```

File: dbxio/delta/table_commands.py [hunk 2]
```diff
@@ -200,6 +215,48 @@ def copy_into_table(
...
+    abs_container_name: str,
+    include_files_pattern: bool = False,
+    replace: bool = False,
+    force_schema: bool = True,
+) -> None:
+    """
+    Copy data from blob storage as a table. All files that match the pattern *.{table_format} will be copied.
+    If force_schema == False it will use schemaHints instead of schema option
+    """
+    create_query = _create_table_query(table, replace, False, False)
```
Comment (NikitaYurasov) [inline dbxio/delta/table_commands.py:233]: let's use kwargs for False, False

File: dbxio/delta/table_commands.py [hunk 3]
```diff
@@ -200,6 +215,48 @@ def copy_into_table(
...
+        if force_schema:
+            options['schema'] = sql_schema
+        else:
+            options['schemaHints'] = sql_schema
+    else:
+        options['mergeSchema'] = 'true'
+
+    options_query = ',\n'.join([f'{k} => {v}' for k, v in options.items()])
+    select_query = dedent(f"""
+    AS SELECT *
```
Comment (NikitaYurasov) [inline dbxio/delta/table_commands.py:250]: function `read_files` sorts output columns in alphabetic order, so insert data can fail bc columns order mismatch

Comment (NikitaYurasov) [reply dbxio/delta/table_commands.py:250]: It's better to replace `*` with `', '.join(table.schema.columns)`

**Review by NikitaYurasov — CHANGES_REQUESTED** [review]

File: dbxio/delta/table_commands.py (hunk 1 above)
Comment (NikitaYurasov) [reply dbxio/delta/table_commands.py:53]: Let's force even schema existence here bc we can face some problems on write without explicit schema declaration

Comment (azorej) [reply dbxio/delta/table_commands.py:53]: We can't force schema for `Create Table As Select (CTAS) statement`

**Review by NikitaYurasov — APPROVED** [review]
LGTM

**Pull request merged**